src/lib/logical_query_plan/jit_aware_lqp_translator.cpp

```diff
@@ -0,0 +1,273 @@
+
+    /* Binary operators */
+    case ExpressionType::Addition:
+    case ExpressionType::Subtraction:
+    case ExpressionType::Multiplication:
+    case ExpressionType::Division:
+    case ExpressionType::Modulo:
+    case ExpressionType::Power:
+    case ExpressionType::Equals:
+    case ExpressionType::NotEquals:
+    case ExpressionType::LessThan:
+    case ExpressionType::LessThanEquals:
+    case ExpressionType::GreaterThan:
+    case ExpressionType::GreaterThanEquals:
+    case ExpressionType::Like:
+    case ExpressionType::NotLike:
+    case ExpressionType::And:
+    case ExpressionType::Or:
+      left = _translate_to_jit_expression(*lqp_expression.left_child(), jit_source, input_node);
+      right = _translate_to_jit_expression(*lqp_expression.right_child(), jit_source, input_node);
+      return std::make_shared<JitExpression>(left, lqp_expression.type(), right, jit_source.add_temporary_value());
+
+    /* Unary operators */
+    case ExpressionType::Not:
+    case ExpressionType::IsNull:
+    case ExpressionType::IsNotNull:
+      left = _translate_to_jit_expression(*lqp_expression.left_child(), jit_source, input_node);
+      return std::make_shared<JitExpression>(left, lqp_expression.type(), jit_source.add_temporary_value());
+
+    default:
+      Fail("Unexpected expression type.");
+  }
+}
+
+std::shared_ptr<const JitExpression> JitAwareLQPTranslator::_translate_to_jit_expression(
+    const LQPColumnReference& lqp_column_reference, JitReadTuple& jit_source,
+    const std::shared_ptr<AbstractLQPNode>& input_node) const {
+  const auto column_id = input_node->find_output_column_id(lqp_column_reference);
+  const auto stored_table_node = std::dynamic_pointer_cast<const StoredTableNode>(lqp_column_reference.original_node());
+  const auto projection_node = std::dynamic_pointer_cast<const ProjectionNode>(lqp_column_reference.original_node());
+
+  if (column_id && stored_table_node) {
+    // we reached a "raw" data column and register it as an input column
+    const auto original_column_id = lqp_column_reference.original_column_id();
+    const auto table_name = stored_table_node->table_name();
+    const auto table = StorageManager::get().get_table(table_name);
+    const auto data_type = table->column_data_type(original_column_id);
+    const auto is_nullable = table->column_is_nullable(original_column_id);
+    const auto tuple_value = jit_source.add_input_column(data_type, is_nullable, column_id.value());
+    return std::make_shared<JitExpression>(tuple_value);
+  } else if (projection_node) {
+    // if the LQPColumnReference references a computed column, we need to compute that expression as well
+    const auto lqp_expression = projection_node->column_expressions()[lqp_column_reference.original_column_id()];
+    return _translate_to_jit_expression(*lqp_expression, jit_source, input_node);
+  } else {
+    Fail("Could not get column from anywhere.");
```

Comment:
That means you can't deal with columns created by an Aggregate expression. Will this Fail get reached if a column from an AggregateNode is referenced?